### PR TITLE
fail more gracefully when using ledgers that reference invalid cases

### PR DESCRIPTION
**Pull request by czue**

### CODE REVIEW

File: corehq/apps/commtrack/tests/test_xml.py [hunk 1]
```diff
@@ -362,6 +362,26 @@ def test_blank_product_id(self):
 
 class BugSubmissionsTest(CommTrackSubmissionTest):
 
+    def test_submit_bad_case_id(self):
+        instance_id = uuid.uuid4().hex
+        amounts = [(p._id, float(i*10)) for i, p in enumerate(self.products)]
```
Comment (dimagimon) [inline corehq/apps/commtrack/tests/test_xml.py]: E226 missing whitespace around arithmetic operator

**Comment by czue** [comment]
@dannyroberts, wait for tests

**Comment by czue** [comment]
:white_check_mark:

**Comment by czue** [comment]
@TylerSheffels

**Comment by TylerSheffels** [comment]
:+1: from me, will let Danny merge

**Pull request merged**